source/draw/RasterMonochrome.ooc

```diff
@@ -155,6 +155,35 @@ RasterMonochrome: class extends RasterPacked {
 			source += sourceWidth
 		}
 	}
+	getScharrDerivative: func(region: IntBox2D, imageX, imageY: FloatImage) {
+		step := 2
+		sourceWidth := this size width
+
+		_pointer := this buffer pointer + region leftTop y * sourceWidth  // this getValue [x,y]
+		_pointerP1 := _pointer + sourceWidth * step
+		_pointerM1 := _pointer - sourceWidth * step
+		destinationX := imageX pointer
+		destinationY := imageY pointer
+
+		for (y in (region leftTop y)..(region rightBottom y)) {
+			for (x in (region leftTop x)..(region rightBottom x)) {
+
+				rightUp := _pointerM1[x + step]
+				rightDown := _pointerP1[x + step]
+				leftUp := _pointerM1[x - step]
+				leftDown := _pointerP1[x - step]
+
+				destinationX@ = ((1.0f * (_pointer[x + step] - _pointer[x - step]) * 0.4641f) + (1.0f * ((rightUp + rightDown) - (leftUp + leftDown)) * 0.26795f))
```

Comment:
@simonmika I don't know, what do you think? Doesn't look sensitive to me, but if it is, now's the time to move it.